src/hotspot/share/gc/g1/g1YoungGCPostEvacuateTasks.cpp

```diff
@@ -182,10 +182,6 @@ class G1FreeHumongousRegionClosure : public HeapRegionClosure {
   }
 
   virtual bool do_heap_region(HeapRegion* r) {
-    if (!r->is_starts_humongous()) {
-      return false;
-    }
```

Comment:
I think this check is a fast-path to quickly filter out definitely uninteresting regions, which will be nearly all of them.  I'm not convinced it should be removed.

Reply:
It's the same argument as [JDK-8292858](https://bugs.openjdk.org/browse/JDK-8292858); the attr-table can (or should) be used instead of the region-type.

Reply:
Yeah, and I thought about objecting to that one on the same grounds.  In retrospect I regret not doing so.

I see no issue with not using the attr-table.  The attr-table is not the UR-source of the information.  It is instead a cache of a non-trivial calculation.  Avoiding it by directly using one of the inputs to that calculation because doing so gives us a fast-path for what is by far the most common case seems perfectly fine.